### PR TITLE
[dependencies]: remove `util/macros`

### PR DESCRIPTION
**Pull request by niklasad1**
Closing #11478

Nothing very exciting in this PR to summarize:
- Remove `util/macros` crate, move `flush!` and `flushln!` to `ethcore/json_tests`.
- Rename: flush, flushln -> flushed_write, flushed_writeln (those were misleading)
- Replace `macros` with `maplit`
- Some changes because `slice_into` and `vec_into` are removed

Note, I didn't replace `flush, flushln` with logging in the json-tests. Let's keep out of this PR

### CODE REVIEW

File: ethcore/src/json_tests/macros.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	($fmt:expr) => (flushed_write!(concat!($fmt, "\n")));
+	($fmt:expr, $($arg:tt)*) => (flushed_write!(concat!($fmt, "\n"), $($arg)*));
+}
+
+/// Write to stdout and flush (ignores errors)
+//
+// TODO(niklasad1): use expect or unwrap only used for tests?!
+#[doc(hidden)]
+pub fn write_and_flush(s: String) {
+	let _ = std::io::Write::write_all(&mut std::io::stdout(), s.as_bytes());
```
Comment (niklasad1) [inline ethcore/src/json_tests/macros.rs]: Changed this to `write_all` instead of `write` seems better because we are not notified if only a part of the buffer is written, thoughts?

File: util/migration-rocksdb/Cargo.toml [hunk 2]
```diff
@@ -2,12 +2,14 @@
 name = "migration-rocksdb"
 version = "0.1.0"
 authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
```
Comment (niklasad1) [inline util/migration-rocksdb/Cargo.toml]: sorry I couldn't resist, it was such as simple fix.

File: util/migration-rocksdb/src/lib.rs [hunk 3]
```diff
@@ -331,7 +324,8 @@ impl Progress {
 		self.current += 1;
 		if self.current == self.max {
 			self.current = 0;
-			flush!(".");
+			// TODO(niklasad1): check why we wrote `.` to stdout before seems needless
```
Comment (niklasad1) [inline util/migration-rocksdb/src/lib.rs]: thoughts?

Comment (dvdplm) [reply util/migration-rocksdb/src/lib.rs]: It's like a poor man's progress indicator I think, at least that's how I read it. Imo it can go.

**Comment by dvdplm** [comment]
(what's up with the force-push, makes it hard to review additions)

File: ethcore/src/json_tests/difficulty.rs [hunk 4]
```diff
@@ -37,8 +37,7 @@ pub fn json_difficulty_test<H: FnMut(&str, HookType)>(
 	for (name, test) in tests.into_iter() {
 		start_stop_hook(&name, HookType::OnStart);
 
-		flush!("   - {}...", name);
-		println!("   - {}...", name);
```
Comment (niklasad1) [inline ethcore/src/json_tests/difficulty.rs]: redundant, but should be `write`!

**Comment by niklasad1** [comment]
> (what's up with the force-push, makes it hard to review additions)

Sorry, my bad it just think is so ugly will `fix missing file` but will merge from now on. Didn't think you would review it so fast.

File: ethcore/engines/authority-round/src/lib.rs [hunk 5]
```diff
@@ -1277,22 +1275,20 @@ impl Engine for AuthorityRound {
...
-				if let Ok(empty_steps) = header_empty_steps(header).as_ref() {
-					format!("[{}]",
-						empty_steps.iter().fold(
-							"".to_string(),
-							|acc, e| if acc.len() > 0 { acc + ","} else { acc } + &e.to_string()))
-
-				} else {
-					"".into()
-				};
+			let empty_steps = header_empty_steps(header).as_ref().map_or(String::new(), |empty_steps| {
```
Comment (niklasad1) [inline ethcore/engines/authority-round/src/lib.rs]: this change is also unrelated, I found this piece very hard to read especially the `fold`

File: ethcore/src/json_tests/macros.rs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+/// Similar to `println!` but flushes stdout in order to ensure the output is emitted immediately.
+#[macro_export]
+macro_rules! flushed_writeln {
+	($fmt:expr) => (flushed_write!(concat!($fmt, "\n")));
+	($fmt:expr, $($arg:tt)*) => (flushed_write!(concat!($fmt, "\n"), $($arg)*));
+}
+
+/// Write to stdout and flush (ignores errors)
+//
+// TODO(niklasad1): use expect or unwrap only used for tests?!
```
Comment (ordian) [inline ethcore/src/json_tests/macros.rs]: I think it's fine if printing to a console fails for some reason

File: util/migration-rocksdb/src/lib.rs (hunk 3 above)
Comment (ordian) [reply util/migration-rocksdb/src/lib.rs]: exactly, we can use a proper progress bar lib like https://github.com/mitsuhiko/indicatif or whatever cool kids are using these days

Comment (niklasad1) [reply util/migration-rocksdb/src/lib.rs]: It was not used, so I removed it

File: rpc/src/v1/types/sync.rs [hunk 7]
```diff
@@ -240,9 +240,11 @@ mod tests {
 	fn test_serialize_transaction_stats() {
 		let stats = TransactionStats {
 			first_seen: 100,
-			propagated_to: map![
-				H512::from_low_u64_be(10) => 50
-			],
+			propagated_to: {
+				let mut map = BTreeMap::new();
```
Comment (ordian) [inline rpc/src/v1/types/sync.rs]: why not a `btreemap!`?

Comment (niklasad1) [reply rpc/src/v1/types/sync.rs]: nit, I think I started to investigate if we could get rid of `map` completely but forgot this.

Comment (dvdplm) [reply rpc/src/v1/types/sync.rs]: I thought you left it on purpose. Why use a macro when it doesn’t save much typing and doesn’t make the code clearer?

Comment (niklasad1) [reply rpc/src/v1/types/sync.rs]: I was just too lazy,  my rule of thumb is when we have more than 2-3 items in a BTreeMap then I use `btreemap!` but then we should clean up the entire crate.

At least after this change we are consistent in the crate but I saw that we can easily remove `maplit` in the `rpc crate`
